javascript/__tests__/do_username.test.js

```diff
@@ -0,0 +1,289 @@
+		const subjectMock = require('../lib/do_username');
+		expect(subjectMock.generate().endsWith('Walrus')).toBeTruthy();
+	});
+});
+
+describe('with descriptor part', () => {
+	it('starts with a descriptor', () => {
+		jest.doMock('../lib/constants', () => {
+			const actual = jest.requireActual('../lib/constants');
+
+			return {
+				...actual,
+				SEA_CREATURES: [],
+				DESCRIPTORS: ['cute'],
+			};
+		});
+
+		// import again after mock
+		const subjectMock = require('../lib/do_username');
+		expect(subjectMock.generate().startsWith('Cute')).toBeTruthy();
+	});
+});
+
+describe('with color part', () => {
+	it('contains a color in generated username', () => {
+		jest.doMock('../lib/constants', () => {
+			const actual = jest.requireActual('../lib/constants');
+
+			return {
+				...actual,
+				COLORS: ['blue'],
+			};
+		});
+		// import again after mock
+		const subjectMock = require('../lib/do_username');
+
+		expect(subjectMock.generate().includes('Blue')).toBeTruthy();
+	});
+});
+
+describe('with maxSize argument', () => {
+	beforeEach(() => {
+		jest.doMock('../lib/constants', () => {
+			const actual = jest.requireActual('../lib/constants');
+
+			return {
+				...actual,
+				SEA_CREATURES: [],
+				DESCRIPTORS: ['cute'],
+				COLORS: ['red'],
+				SEA_LIST: ['walrus'],
+			};
+		});
+	});
+
+	it('responds with username shorter than or equal to given size', () => {
+		// import again after mock
+		const subjectMock = require('../lib/do_username');
+		expect(subjectMock.generate(15).length).toBeLessThanOrEqual(15);
+	});
+
+	it('responds with full combination when appropriate', () => {
+		// import again after mock
+		const subjectMock = require('../lib/do_username');
+		expect(subjectMock.generate(100)).toBe('CuteRedWalrus');
+	});
+
+	it('responds with descriptor + noun when appropriate', () => {
+		// import again after mock
+		const subjectMock = require('../lib/do_username');
+		expect(subjectMock.generate(12)).toBe('CuteWalrus');
+	});
+
+	it('responds with color + noun when appropriate', () => {
+		// import again after mock
+		const subjectMock = require('../lib/do_username');
+		expect(subjectMock.generate(9)).toBe('RedWalrus');
+	});
+
+	it('responds with part of the noun when appropriate', () => {
+		// import again after mock
+		const subjectMock = require('../lib/do_username');
+		expect(subjectMock.generate(5)).toBe('Walru');
+	});
+});
+
+describe('when is invalid', () => {
+	const argumentError = 'The maxSize argument must be an integer number greater than zero.';
+	it('raises ArgumentError negative values', () => {
+		expect(() => {
+			subject.generate(-99);
+		}).toThrow(argumentError);
+	});
+
+	it('raises ArgumentError negative values', () => {
+		expect(() => {
+			subject.generate(0);
+		}).toThrow(argumentError);
+	});
+
+	it('raises ArgumentError negative values', () => {
+		// import again after mock
+		expect(() => {
+			subject.generate('abc');
+		}).toThrow(argumentError);
+	});
+});
```

Comment:
These should all be within the `generate` describe block

Reply:
done